Resync WPT streams API tests from upstream

youennf opened the pull request:
#### beacb48b7ebe764fac7b7436b4af621704ebc847
<pre>
Resync WPT streams API tests from upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=248271">https://bugs.webkit.org/show_bug.cgi?id=248271</a>
rdar://problem/102624488

Reviewed by Tim Nguyen.

Resynced up to e97fac4791931fb7455ba3fad759d362c7108b09.

* LayoutTests/imported/w3c/web-platform-tests/common/gc.js: Added.
(self.garbageCollect.async if):
(self.garbageCollect.gcRec):
(self.garbageCollect):
* LayoutTests/imported/w3c/web-platform-tests/streams: Resynced.

Canonical link: <a href="https://commits.webkit.org/259205@main">https://commits.webkit.org/259205@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fc58883985cf5cc601df80ff9828f0b8021ae215

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104349 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13434 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37260 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113565 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173858 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14529 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4349 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96573 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112605 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110117 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11194 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94256 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38827 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93057 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25867 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80497 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6794 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27224 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6924 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3778 "Found 1 new test failure: fast/images/avif-image-document.html (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12946 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46782 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6346 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8714 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Please add the upstream revision in commit message

youennf committed:
Resync WPT streams API tests from upstream
https://bugs.webkit.org/show_bug.cgi?id=248271
rdar://problem/102624488

Reviewed by Tim Nguyen.

Resynced up to e97fac4791931fb7455ba3fad759d362c7108b09.

* LayoutTests/imported/w3c/web-platform-tests/common/gc.js: Added.
(self.garbageCollect.async if):
(self.garbageCollect.gcRec):
(self.garbageCollect):
* LayoutTests/imported/w3c/web-platform-tests/streams: Resynced.

Canonical link: https://commits.webkit.org/259205@main
